app/lib/backup/sender.ts

```diff
@@ -0,0 +1,105 @@
+import { Client as StorachaClient, SpaceDID } from "@storacha/ui-react"
+import { JobID, JobRequest } from "@/api"
+import { LaunchParams } from "@telegram-apps/sdk-react"
+import { Session, StringSession } from "@/vendor/telegram/sessions"
+import {  Principal } from '@ipld/dag-ucan'
+import * as SpaceBlob from '@storacha/capabilities/space/blob'
+import * as SpaceIndex from '@storacha/capabilities/space/index'
+import * as Upload from '@storacha/capabilities/upload'
+import { NameView } from "@storacha/ucn/api"
+
+
+// default to 1 hour
+const defaultDuration = 1000 * 60 * 60
+
+const CURRENT_VERSION = "1";
+
+export interface Context {
+  storacha: StorachaClient
+  servicePrincipal: Principal,
+  name: NameView
+  spaceDID: SpaceDID
+  encryptionPassword: string
+  session: Session
+  launchParams: LaunchParams
+  sendRequest: (jr: JobRequest) => Promise<void>
+}
+
+export const create = async (ctx: Context) => {
+  return new JobSender(ctx)
+}
+
+class JobSender {
+  #spaceDID
+  #name
+  #encryptionPassword
+  #session
+  #launchParams
+  #storacha
+  #servicePrincipal
+  #sendRequest
+
+  constructor({ spaceDID, name, encryptionPassword, session, launchParams, storacha, servicePrincipal, sendRequest} : Context) {
+    this.#spaceDID = spaceDID
+    this.#name = name
+    this.#encryptionPassword = encryptionPassword
+    this.#session = session
+    this.#launchParams = launchParams
+    this.#storacha = storacha
+    this.#servicePrincipal = servicePrincipal
+    this.#sendRequest = sendRequest
+  }
+
+  #saveSessionToString() {
```

Comment:
why not using `session.save()`? [more here](https://gram.js.org/getting-started/authorization#string-session)